packages/react-slider/src/components/Slider/useSliderState.tsx

```diff
@@ -0,0 +1,238 @@
+import * as React from 'react';
+import { useId, useControllableState, useMount } from '@fluentui/react-utilities';
+import { SliderSlots, SliderState, SliderCommon } from './Slider.types';
+
+/**
+ * Validates that the `value` is a number and falls between the min and max.
+ *
+ * @param value - the value to be clamped
+ * @param min - the lowest valid value
+ * @param max - the highest valid value
+ */
+const clamp = (value: number, min: number, max: number): number => Math.max(min, Math.min(max, value || 0));
+
+/**
+ * Gets the current percent of specified value between a min and max
+ *
+ * @param value - the value to find the percent
+ * @param min - the lowest valid value
+ * @param max - the highest valid value
+ */
+const getPercent = (value: number, min: number, max: number) => {
+  return max === min ? 0 : ((value - min) / (max - min)) * 100;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const on = (element: Element, eventName: string, callback: (ev: any) => void) => {
```

Comment:
Isn't this already defined elsewhere? Or if that's only in v8, you should look at how other converged components are handling events.